components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImpl.java

```diff
@@ -644,4 +679,34 @@ private AuthzCodeDO createAuthzCodeDo(String consumerKey, String authorizationKe
                 pkceCodeChallengeMethod);
     }
 
+    private String getAuthenticatedIDP(AuthenticatedUser user) {
+
+        String authenticatedIDP;
```

Comment:
How about a debug log to indicate how we resolved the idp name and the resolved idp name?

Reply:
+1 will add